src/main/java/org/jboss/threads/ThreadLocalResettingRunnable.java

```diff
@@ -19,22 +25,32 @@ public String toString() {
     }
 
     static final class Resetter {
-        private static final long threadLocalMapOffs;
-        private static final long inheritableThreadLocalMapOffs;
+        private static final MethodHandle threadLocalMapHandle;
+        private static final MethodHandle inheritableThreadLocalMapHandle;
 
         static {
+            MethodHandles.Lookup threadLookup;
             try {
-                threadLocalMapOffs = JBossExecutors.unsafe.objectFieldOffset(Thread.class.getDeclaredField("threadLocals"));
-                inheritableThreadLocalMapOffs = JBossExecutors.unsafe.objectFieldOffset(Thread.class.getDeclaredField("inheritableThreadLocals"));
+                threadLookup = MethodHandles.privateLookupIn(Thread.class, MethodHandles.lookup());
+                threadLocalMapHandle = threadLookup.unreflectVarHandle(Thread.class.getDeclaredField("threadLocals")).toMethodHandle(VarHandle.AccessMode.SET).asType(MethodType.methodType(void.class, Object.class));
+                inheritableThreadLocalMapHandle = threadLookup.unreflectVarHandle(Thread.class.getDeclaredField("inheritableThreadLocals")).toMethodHandle(VarHandle.AccessMode.SET).asType(MethodType.methodType(void.class, Object.class));
+            } catch (IllegalAccessException e) {
+                throw new IllegalAccessError(e.getMessage());
             } catch (NoSuchFieldException e) {
                 throw new NoSuchFieldError(e.getMessage());
             }
         }
 
         static void run() {
             final Thread thread = Thread.currentThread();
-            JBossExecutors.unsafe.putObject(thread, threadLocalMapOffs, null);
-            JBossExecutors.unsafe.putObject(thread, inheritableThreadLocalMapOffs, null);
+            try {
```

Comment:
We gotta ask @galderz how this perform in native

Reply:
I expect it to be OK: AFAIK, VarHandle is using these method handles internally, and we're acquiring them in static init. My expectation is that the calls will be inlined.

However, either way it's not going to work without `--add-opens java.base/java.lang=org.jboss.threads` or `...=ALL-UNNAMED`. That's the real sticky part.